Core/Api/BaseApiClient.cs

```diff
@@ -25,7 +25,26 @@ public void AddToken(string token)
 
     public RestResponse Execute(RestRequest request)
     {
-        LogSession.CurrentSession.Debug($"Resource: '{request.Resource}' Method: '{request.Method}' " +
+        var urlSegments = request.Parameters
+            .Where(p => p.Type == ParameterType.UrlSegment)
+            .Where(x => !string.IsNullOrEmpty(x.Name)
+                    && !string.IsNullOrEmpty(x.Value?.ToString()));
+        var queryStrings = request.Parameters
+            .Where(p => p.Type == ParameterType.QueryString)
+            .Where(x => !string.IsNullOrEmpty(x.Name)
+                    && !string.IsNullOrEmpty(x.Value?.ToString()));
+        var resource = request.Resource;
+
+        foreach (var urlSegment in urlSegments)
+        {
+            resource = resource.Replace($"{{{urlSegment.Name}}}", urlSegment.Value?.ToString());
+        }
+        foreach (var queryString in queryStrings)
+        {
+            resource += "?" + queryString.ToString();
+        }
```

Comment:
This should work if you 1) add the token to the resource url and 2) specify ParameterType.UrlSegment like this:

var client = new RestClient("http://localhost");
var request = This should work if you 1) add the token to the resource url and 2) specify ParameterType.UrlSegment like this:

var client = new RestClient("http://localhost");
var request = new RestRequest("resource?auth_token={authToken}", Method.POST);
request.AddParameter("auth_token", "1234", ParameterType.UrlSegment);    
request.AddBody(json);
var response = client.Execute(request); RestRequest("resource?auth_token={authToken}", Method.POST);
request.AddParameter("auth_token", "1234", ParameterType.UrlSegment);    
request.AddBody(json);
var response = client.Execute(request);


or

var request = new RestRequest("resource?auth_token={token}", Method.POST);
request.AddUrlSegment("token", "1234");
Alternatively, you can add a parameter without a template:

var request = new RestRequest("resource", Method.POST);
request.AddQueryParameter("auth_token", "1234); 
or

var request = new RestRequest("resource", Method.POST);
request.AddParameter("auth_token", "1234", ParameterType.QueryString);

Reply:
Here I did not add requests to the URL, but logged them. If this is removed, the URL without parameters is returned in "request.Resource".